packages/services/f-perf/src/index.js

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable no-unused-vars */
+import Perfume from 'perfume.js';
+
+/**
+* @overview Vue Plugin logging Real User Metrics
+* By default will push to `window.dataLayer`
+* Custom logger function can be used
+* Builds on top of PerfumeJS https://github.com/zizzamia/perfume.js
+*
+* @module f-perf
+*/
+
+/**
+ * Default Performance log Function, pushes metrics to GTM dataLayer
+ * @param {String} metricName Metric logged
+ * @param {Object} data Metric data
+ */
+let logPerformance = (metricName, data) => {
+    window.dataLayer = window.dataLayer || [];
+    return window.dataLayer.push({
+        event: 'trackEvent',
+        eventCategory: 'engagement',
+        eventAction: 'app_performance',
+        eventLabel: metricName,
+        eventValue: metricName === 'cls' ? data * 1000 : data,
+        nonInteraction: true
+    });
+};
+
+/**
+ * Instantiates PerfumeJS with some of the defaults
+ */
+const perfumeJS = new Perfume({
+    analyticsTracker: options => {
+        const {
+            metricName,
+            data,
+            eventProperties,
+            navigatorInformation,
+            vitalsScore
+        } = options;
+
+        switch (metricName) {
```

Comment:
Not a fan of the giant switch statement, could we instead store an object who's keys represent the correct metric name? a look up table of some kind. Example

```
const lookUpTable = {
    'navigationTiming': 'navigationTiming',
    'firstContentfulPaint': 'fcp'
};

logPerformance(Object.keys(lookUpTable).find(key => lookUpTable[key] === metricName), data)
```

Reply:
There is certainly scope for that. I was considering similar approach but enabling `options` to include specific metrics tracked. There has been concern on how much logging will be sent to GTM if every vertical or component starts logging.

Reply:
I don't really mind how we optimise the switch statement, I was just giving a suggestion, happy with another approach if it reduces the repetition of code. My only thoughts around a lookup table was that would enable us to keep an eye on what event types were being submitted to the logging server, as they would be controlled with PR's being raised to add additional types. As for the use of GTM as our logging platform, I have to agree I don't think that is a best use case for GTM, would we not be better off using something like sentry? @tommcclean I was under this impression we were trialling sentry atm?

Reply:
Heard about sentry being mentioned. GTM could provide a different dimension as the data is larger and could potentially be combined to provide a better picture.

Reply:
A couple of concerns for me with GTM use. 

1) Metrics tell us nothing meaningful about the user, they either report errors and stack traces or they provide information that is relevant to performance on a global level. GTM seems like a user specific event tracking system.
2) Events use up monthly quotas so we would want to be sure that letting people lose with this would not dramatically cost us anything as a result of the potential significant increase in events being fired. (don't know what we currently get)

I may be alone in this thought though so I think it would be better discussed at a wider group level?

Reply:
Absolutely, this is a wider group discussion. I wanted to gauge earlier version views and get the sense of what other may have in mind. 

Thank you so much in contributing @matthewhardern. I am happy to have that open conversation going on here, but will follow up with a Guild meeting before we can go ahead with these changes.

Reply:
I have added a slot in the guild to discuss it tomorrow.